### PR TITLE
2.x Discovery endpoint

### PR DESCRIPTION
**Pull request by e0ipso**
This was missing, but no more!

This PR will add testing to the discovery endpoint and add in the missing features for the endpoint.

### CODE REVIEW

File: src/Plugin/resource/DataProvider/DataProviderPlug.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+   * {@inheritdoc}
+   */
+  public function update($identifier, $object, $replace = FALSE) {
+    throw new NotImplementedException('You cannot create plugins through the API.');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function remove($identifier) {
```
Comment (amitaibu) [inline src/Plugin/resource/DataProvider/DataProviderPlug.php:124]: this looks like a security issue

Comment (amitaibu) [inline src/Plugin/resource/DataProvider/DataProviderPlug.php:124]: @mateu-aguilo-bosch ^^

Comment (e0ipso) [inline src/Plugin/resource/DataProvider/DataProviderPlug.php:124]: This has been fixed via access check for the write controllers.

**Comment by e0ipso** [comment]
I merged #566 in here to address your concern and to add additional testing.

**NEW FEATURE**: Privileged users can enable / disable resources using the API. How cool is that?

File: src/Plugin/resource/DataProvider/DataProviderPlug.php (hunk 1 above)
Comment (amitaibu) [inline src/Plugin/resource/DataProvider/DataProviderPlug.php:124]: can you point me to the code - I'm not clear on what you mean?

Comment (e0ipso) [inline src/Plugin/resource/DataProvider/DataProviderPlug.php:124]: https://github.com/RESTful-Drupal/restful/pull/565/files#diff-ebd241a9a98fd5b56a0b645bc1ce949dR119

Comment (e0ipso) [inline src/Plugin/resource/DataProvider/DataProviderPlug.php:124]: Testing is added in: https://github.com/RESTful-Drupal/restful/pull/565/files#diff-e12329e5922ad8f5be15b2261f290cfeR94

File: tests/RestfulDataProviderPlugPluginsTestCase.test [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    $handler->setPath('articles:1.0');
+    try {
+      $handler->process();
+      $this->fail('Un-privileged user can disable endpoints.');
+    }
+    catch (ForbiddenException $e) {
+      $this->pass('Un-privileged user cannot disable endpoints.');
+    }
+
+    // Make sure that unauthorized users can enable/disable resources via the
```
Comment (amitaibu) [inline tests/RestfulDataProviderPlugPluginsTestCase.test]: unauthorized => authorized

File: src/Plugin/resource/DataProvider/DataProviderPlug.php (hunk 1 above)
Comment (amitaibu) [inline src/Plugin/resource/DataProvider/DataProviderPlug.php:124]: :+1:

Comment (amitaibu) [inline src/Plugin/resource/DataProvider/DataProviderPlug.php:124]: feel free to merge :)

Comment (e0ipso) [inline src/Plugin/resource/DataProvider/DataProviderPlug.php:124]: :+1:

Some (unrelated) tests came back red. I'll fix them before merging.

**Comment by e0ipso** [comment]
w00t! Tests are green!

Merging.

**Pull request merged**